surefire-api/src/main/java/org/apache/maven/surefire/api/util/DefaultRunOrderCalculator.java

```diff
@@ -181,14 +143,55 @@ public void addTestToOrders( String className, LinkedHashMap<String, List<String
         }
     }
 
+    public TestListResolver getTestListResolver()
+    {
+        String orderParam = parseTestOrder( System.getProperty( "test" ) );
+        if ( orderParam == null  )
+        {
+            throw new IllegalStateException( "Please set system property -Dtest to use fixed order"  );
+        }
+        List<String> list = Arrays.asList( orderParam.split( "," ) );
+        return new TestListResolver( list );
+    }
+
+    public String[] getClassAndMethod( String request )
+    {
+        String[] classAndMethod = { request, request };
+        if ( request.contains( "(" ) )
+        {
+            String[] nameSplit1 = request.split( "\\(" );
+            classAndMethod[0] = nameSplit1[1].substring( 0, nameSplit1[1].length() - 1 );
+            classAndMethod[1] = nameSplit1[0];
+        }
+        return classAndMethod;
+    }
+
+    private RunOrder getRunOrderMethod()
+    {
+        if ( runOrder.length == 0 )
+        {
+            throw new IllegalStateException( "Expected runOrder to be testorder." );
```

Comment:
If no runOrders are provided (e.g., if you omit -Dsurefire.runOrder), would runOrder.length be 0 or would it be 1 (the default runOrder should be filesystem)?

Anyway, I don't think you should be throwing an exception if runOrder.length is 0. If there is a testorder in runOrder then return true, otherwise just return false. You need not return testorder from searching through the array and then checking for testorder again.

Reply:
If no runOrders are provided, runOrder.length will be 1, so if you check if there is a testorder in runOrder it will always return true

Reply:
Fine, you can throw an exception then but the message should not be that the runOrder is expected to be testorder. Instead, it should just say that runOrder is not expected to ever be 0 since a default value should be used if no runOrder is provided

Reply:
I will delete it since Line is longer than 120 characters. This message is too long